src/rendering/vcLabelRenderer.cpp

```diff
@@ -1,40 +1,142 @@
 #include "vcLabelRenderer.h"
 #include "udPlatformUtil.h"
+#include "gl/vcMesh.h"
+#include "gl/vcRenderShaders.h"
+#include "vcInternalModels.h"
 
 #include "imgui.h"
+#include <gl\vcGLState.h>
 
-bool vcLabelRenderer_Render(vcLabelInfo *pLabelRenderer, const udDouble4x4 &viewProjectionMatrix, const udUInt2 &screenSize)
+static struct vcLabelShader
+{
+  vcShader *pShader;
+  vcShaderConstantBuffer *pEveryObjectConstantBuffer;
+
+  vcShaderSampler *pDiffuseSampler;
+
+  struct
+  {
+    udFloat4x4 u_worldViewProjectionMatrix;
+    udFloat4 u_screenSize;
+  } everyObject;
+} gShader;
+
+vcMesh *g_LabelMesh = nullptr;
+
+static int gRefCount = 0;
+udResult vcLabelRenderer_Init()
+{
+  udResult result;
+  gRefCount++;
```

Comment:
This uses postfix, the decrement uses prefix. Additionally, is this intended to be thread safe?

Reply:
No this is not mean to be thread safe. I have fixed the prefix issue.